crates/ntx-builder/src/actor/mod.rs

```diff
@@ -0,0 +1,339 @@
+pub mod account_state;
+mod execute;
+mod inflight_note;
+mod note_state;
+
+use std::sync::Arc;
+
+use account_state::{NetworkAccountState, TransactionCandidate};
+use execute::NtxError;
+use futures::FutureExt;
+use miden_node_proto::domain::account::NetworkAccountPrefix;
+use miden_node_proto::domain::mempool::MempoolEvent;
+use miden_node_utils::ErrorReport;
+use miden_objects::account::Account;
+use miden_objects::block::BlockNumber;
+use miden_objects::transaction::TransactionId;
+use miden_remote_prover_client::remote_prover::tx_prover::RemoteTransactionProver;
+use tokio::sync::{AcquireError, RwLock, Semaphore, mpsc};
+use tokio_util::sync::CancellationToken;
+use url::Url;
+
+use crate::block_producer::BlockProducerClient;
+use crate::builder::ChainState;
+use crate::store::StoreClient;
+
+// ACTOR SHUTDOWN REASON
+// ================================================================================================
+
+/// The reason an actor has shut down.
+pub enum ActorShutdownReason {
+    /// Occurs when an account actor processes a transaction that reverts it's account creation.
+    AccountReverted(NetworkAccountPrefix),
+    /// Occurs when an account actor detects failure in the messaging channel used by the
+    /// coordinator.
+    EventChannelClosed,
+    /// Occurs when an account actor detects failure in acquiring the rate-limiting semaphore.
+    SemaphoreFailed(AcquireError),
+    /// Occurs when an account actor detects its corresponding cancellation token has been triggered
+    /// by the coordinator.
+    Cancelled(NetworkAccountPrefix),
```

Comment:
This is used for graceful shutdown of actors, right? If so, I'd mention this in the comment.